### PR TITLE
Validate IMU, odometry, timing, frame names.

### PR DESCRIPTION
**Pull request by gaschler**
Checks for 5 common mistakes in a bag file listed in #529.
- Difference between serialization and sensor time.
- Wrong IMU measurement format.
- Input contains cartographer output frames.
- Sensor measurements have large gaps in time.
- IMU or odometry have invalid values.

### CODE REVIEW

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc [hunk 1]
```diff
@@ -76,13 +79,69 @@ std::unique_ptr<std::ofstream> CreateTimingFile(const std::string& frame_id) {
   return timing_file;
 }
 
+void CheckImuMessage(const sensor_msgs::Imu& imu_message) {
+  auto linear_acceleration = ToEigen(imu_message.linear_acceleration);
+  if (std::isnan(linear_acceleration.norm()) ||
+      linear_acceleration.norm() < 3. || linear_acceleration.norm() > 30.) {
```
Comment (cschuet) [inline cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: pull out 3 and 30 as constants?

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc [hunk 2]
```diff
@@ -76,13 +79,69 @@ std::unique_ptr<std::ofstream> CreateTimingFile(const std::string& frame_id) {
...
+          << "\". This is almost always output published by cartographer and "
+             "should not appear as input. (Unless you have some complex "
+             "remove_frames configuration, this is will not work. Simplest "
+             "solution is to record input without cartographer running.)";
+    }
+  }
+}
+
+bool IsPointDataType(const std::string& data_type) {
+  const std::set<std::string> point_data_types = {
```
Comment (cschuet) [inline cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: could this not be static somehow?

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc [hunk 3]
```diff
@@ -76,13 +79,69 @@ std::unique_ptr<std::ofstream> CreateTimingFile(const std::string& frame_id) {
...
 void Run(const std::string& bag_filename, const bool dump_timing) {
   rosbag::Bag bag;
   bag.open(bag_filename, rosbag::bagmode::Read);
   rosbag::View view(bag);
 
-  std::map<std::string, PerFrameId> per_frame_id;
+  std::map<std::string, FrameProperties> frame_id_to_properties;
   size_t message_index = 0;
+  int num_imu_messages = 0;
+  double sum_imu_acceleration = 0;
```
Comment (cschuet) [inline cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: 0.

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc [hunk 4]
```diff
@@ -147,19 +216,65 @@ void Run(const std::string& bag_filename, const bool dump_timing) {
...
+        LOG_FIRST_N(WARNING, 1) << stream.str();
+      }
+    }
   }
   bag.close();
 
+  if (num_imu_messages > 0) {
+    double average_imu_acceleration = sum_imu_acceleration / num_imu_messages;
+    if (std::isnan(average_imu_acceleration) ||
+        average_imu_acceleration < 9.5 || average_imu_acceleration > 10.5) {
```
Comment (cschuet) [inline cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: Here and below should we collect these number constants in one place as constants to make it more convenient to change them?

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc (hunk 3 above)
Comment (gaschler) [reply cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: Done

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc (hunk 4 above)
Comment (gaschler) [reply cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: Done

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc (hunk 1 above)
Comment (gaschler) [reply cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: Done

File: cartographer_ros/cartographer_ros/rosbag_validate_main.cc (hunk 2 above)
Comment (gaschler) [reply cartographer_ros/cartographer_ros/rosbag_validate_main.cc]: Done

**Pull request merged**